DOT Cashback from Top Retailers

MissMoran opened the pull request:
#### Project Abstract
Buy with Fiat - Get DOT Cashback for Online Purchases.
Bringing DOT cashback to everyday shopping for Polkadot wallet users.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (DOT Cashback from Top Retailers.md).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (Polkadot AssetHub (USDC & DOT) address in the application and bank details via email, if applicable).
- [x] I understand that an agreed upon percentage of each milestone will be paid in vested DOT, to the Polkadot address listed in the application.
- [x] I am aware that, in order to receive a grant, I (and the entity I represent) have to successfully complete a KYC/KYB check.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

MissMoran commented:
I have read and hereby sign the Contributor License Agreement.

On Sun, Dec 29, 2024 at 1:04 PM github-actions[bot] <
***@***.***> wrote:

> *CLA Assistant Lite bot:* Thank you for your submission, we really
> appreciate it. Like many open source projects, we ask that you sign our Contributor
> License Agreement
> <https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md>
> before we can accept your contribution. Please submit the following text as
> a separate comment:
> ------------------------------
>
> I have read and hereby sign the Contributor License Agreement.
> ------------------------------
>
> You can retrigger this bot by commenting *recheck* in this Pull Request
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2481#issuecomment-2564687843>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BN6YEJ3NIUWJAECME62NY7D2H7JKPAVCNFSM6AAAAABUKWF5HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNRUGY4DOOBUGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

PieWol commented:
Hey @MissMoran ,
thank you for your interest in our grants program. 
Currently there is a similar project in development. https://polkadot.com/app
To my understanding the only difference would be that while you allow users to purchase in fiat, in the PolkadotApp users already pay with DOT. The cashback being in DOT in both cases. So I assume that this little difference wouldn't be enough to  justify funding such a solution twice. I'd appreciate it if you could point out all relevant differences to the app I mentioned, in case that I'm missing something :)
Thanks!

keeganquigley reviewed (changes requested):
Thanks for the application @MissMoran in addition to the above comments, could you please add tables to the milestones as shown in the [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md)? This will ensure consistent formatting and also make it easier to read. Thanks!

keeganquigley commented:
pinging @MissMoran

MissMoran commented:
Hi @keegan.
Thank you for that.
We're working on revising the application according to the new findings and
will submit it tomorrow including the formatting request.


On Tue, Jan 14, 2025 at 9:35 PM Keegan | W3F ***@***.***>
wrote:

> pinging @MissMoran <https://github.com/MissMoran>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2481#issuecomment-2590951381>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BN6YEJ2FQ4IUZEBLAOZ5LED2KVRJTAVCNFSM6AAAAABUKWF5HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJQHE2TCMZYGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley commented:
thanks for the update @MissMoran sounds good, let us know when we should take another look.

MissMoran commented:
Hi @keeganquigley

Just updated our PR with additional info based on your questions and
David's remarks (in yellow).
I also revised the formatting in the milestones as requested.

Adding here the additional info that was added to your convenience:
Are you already working with anyone?
Currently we don't have a Polkadot partner wallet.
One of our milestones is to have a signed partnership agreement.
(we do have signed partnerships on different blockchains)

working on other things other than just cashback to purchase something
(healthy activities/sports?) - a clear differentiation
If so, this would be , and you could focus on this as part of the
application
Our solution offers a broad framework for DOT rewards in general. .
On top of the cashback integration, we're working on 'Task to Earn'
offerings.
DApps can offer DOT rewards on activities, incentivising current and new
investors to Earn more DOT.
The DOT rewards could be implemented in various verticals as Move to Earn
(health & lifestyle), Games, Rewarded Social activities, Data Sharing and
more.

*Comparison: Bring VS Raise*

Technical Overview:

   -
*Diverse & Scalable vs. Single Integration   *Bring: Open source, designed
   to be used by any Polkadot builder
   - Raise: Native integration - limited to one (Polkadot app)

   -
*Architecture *1. Versatile Infrastructure for DOT Rewards: a comprehensive
   infrastructure designed to support various DOT reward types, with cashback
   rewards as the first of many applications.
   2. Our codebase is open source MIT license, enabling security audits,
   comprehensive RESTful API and community contributions.
   3. Our system includes a built-in revenue-sharing mechanism, ensuring a
   seamless revenue cut for the Polkadot builder.

   - *Performance Optimisation mechanism*
AB Testing, Reporting, Analytics

   - *Support Multiple Tokens*
We enable wallets to reward their users with a variety of Polkadot tokens,
   rather than a one limited offer, so the user can choose how they'd like to
   get their rewards

   - *Quick & Easy Integration*
1. A ready made white label iFrame template to be integrated and adjusted
   to the wallet's brand, as quickly as 'Plug-and-Play'
   2. An API for an easy integration.

   - *Diverse User Experience *
Bring's integration can be implemented as DOT cashback or rewards in
   various ways:
   1. Powerful popup offerings while browsing (Chrome extension)
   2. Wallet's portal - users can browse through the offering via a
   dedicated section within the wallet (websites/native apps) - engage with
   the wallet
   3. Organic mobile experience



On Wed, Jan 15, 2025 at 8:18 PM Keegan | W3F ***@***.***>
wrote:

> thanks for the update @MissMoran <https://github.com/MissMoran> sounds
> good, let us know when we should take another look.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2481#issuecomment-2593644371>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BN6YEJ2MAOL4UUEII7MZTQT2K2RABAVCNFSM6AAAAABUKWF5HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJTGY2DIMZXGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

MissMoran commented:
Hi @keegan,

Just making sure you got our last updates.

Thanks.


On Wed, Jan 15, 2025 at 8:18 PM Keegan | W3F ***@***.***>
wrote:

> thanks for the update @MissMoran <https://github.com/MissMoran> sounds
> good, let us know when we should take another look.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2481#issuecomment-2593644371>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BN6YEJ2MAOL4UUEII7MZTQT2K2RABAVCNFSM6AAAAABUKWF5HGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJTGY2DIMZXGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

keeganquigley reviewed (changes requested):
Thanks for your answers @MissMoran I have a few additional comments:

- The address provided seems to be an ETH/EVM address. Could you please provide a DOT address instead? Along with a USDC Asset Hub address?
- Are you willing to accept 50% payment in vested DOT? If so could you indicate this under total costs? See https://github.com/w3f/Grants-Program?tab=readme-ov-file#level_slider-levels
- The grants program focuses mainly on technical development and not so much the business development side of things. Therefore we typically wouldn't fund the deliverables in milestone 3. This includes the designs, as we usually ask for some initial mock-ups to be provided ahead of time. For this reason, I'm wondering if you'd be willing to remove M3 and reduce costs to bring it down to $55k?

keeganquigley commented on applications/DOT Cashback from Top Retailers.md:
```diff
@@ -0,0 +1,289 @@
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT|
+| **0b.** | Documentation | 1. A signed letter by the partner wallet approving the completion of the development 2. Walkthrough Videos of the: Portal + Chrome Extension + Claim process demo mp4 +  Backend API - Components, Functionality of Chrome extension - via Postman + Presenting the UI 3. Detailed Documentation - open source (Chrome extension code, UIs and APIs): API Docs + NPM package for in ...
+| **0c.** | Testing and Testing Guide | a public link showing the DOT cashback feature in the partner wallet|
```
Will there be a user-facing guide as well?

MissMoran commented:
Hi @keeganquigley,

Thanks for your feedback, my answers below:

1. I updated *a new DOT address.*
As indicated in the reference below, there's an option to get the
remaining 50% in fiat - an email with payment details was sent and
indicated in the application with date and time as requested.
I also added the *Asset Hub address for USDC* to your request.

Payment: In case of payment in USDC, please provide a Polkadot AssetHub
address and the currency (e.g. 15oF4... (USDC)).
In the case of fiat payment, please share your bank account privately with
***@***.*** via your contact email (see below) and list here the
date and time of your email (e.g. Fiat 24.12.1971, 11:59).

2. Yes, we agree and aware to the DOT payment and vesting

3. I understand your point.
I excluded this step from the milestones due to the focus on technical
aspects and updated the amount accordingly.

-   **Total Estimated Duration:**  4 months
-   **Full-Time Equivalent (FTE):**  2.5 FTE
-   **Total Costs:**  55,000 USD
-   **DOT %:**  50% payment in vested DOT

4. Indeed, we will provide a user-facing guide in the end as updated here:

| **0c.** | Testing and Testing Guide | 1. A public link showing the DOT
cashback feature in the partner wallet 2. A detailed user-facing guide
showcasing the cashback integration step by step including screenshots and
explanations on the various options|

I hope this covers all your questions.
Thanks.


On Wed, Jan 22, 2025 at 9:28 PM Keegan | W3F ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In applications/DOT Cashback from Top Retailers.md
> <https://github.com/w3f/Grants-Program/pull/2481#discussion_r1925855561>:
>
> > +| **0c.** | Testing and Testing Guide | functional UI demo|
> +| 1. | Partnership Agreement | 1. Identify and engage with a Polkadot wallet partner 2. Sign a collaboration agreement |
> +| 2. | UI Design Development | Work closely with the partner wallet to design all user interfaces, including: 1. Cashback Portal 2. Cashbach Popup (Chrome extension)|
> +
> +### **Milestone 4 — Development, Integration and QA**
> +
> +**Estimated Duration:** 1 month
> +**FTE:** 2
> +**Costs:** $36,000 USD
> + **Overview:** Develop, integrate, and test the DOT cashback system with the partner wallet, ensuring a fully functional product ready for deployment.
> +
> +| Number | Deliverable | Specification |
> +| -----: | ----------- | ------------- |
> +| **0a.** | License | MIT|
> +| **0b.** | Documentation | 1. A signed letter by the partner wallet approving the completion of the development 2. Walkthrough Videos of the: Portal + Chrome Extension + Claim process demo mp4 +  Backend API - Components, Functionality of Chrome extension - via Postman + Presenting the UI 3. Detailed Documentation - open source (Chrome extension code, UIs and APIs): API Docs + NPM package for integrating popup + Cashback Portal Integration docs + UI Breakdown 4.User guides to the partner wallet for testing: QA 1_User guide for testing + QA 2_SOP |
> +| **0c.** | Testing and Testing Guide | a public link showing the DOT cashback feature in the partner wallet|
>
> Will there be a user-facing guide as well?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/w3f/Grants-Program/pull/2481#pullrequestreview-2568095919>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BN6YEJ43LCDI6MVOYYTEIX32L7WMZAVCNFSM6AAAAABUKWF5HGVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKNRYGA4TKOJRHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

MissMoran commented:
@keeganquigley I replied to your email with all the additional info, but i see now the PR has failed to update due to other review requesting changes. Informing I'm looking into that.

MissMoran commented:
Hi @keeganquigley,
Trying to update our md with no success. 
We have reviewed your requests for changes and applied them. 
Pleasde advise.

PieWol commented:
Hey @MissMoran ,
The PR is fine. No need for any immediate changes. We will discuss it internally and get back to you soon.
